test/functional/mempool_limit.py

```diff
@@ -20,55 +23,76 @@ def set_test_params(self):
         ]]
         self.supports_cli = False
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
         txouts = gen_return_txouts()
-        relayfee = self.nodes[0].getnetworkinfo()['relayfee']
+        # Helper function that generates large transactions
+
+        node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        relayfee = node.getnetworkinfo()['relayfee']
 
         self.log.info('Check that mempoolminfee is minrelytxfee')
-        assert_equal(self.nodes[0].getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
-        assert_equal(self.nodes[0].getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
+        assert_equal(node.getmempoolinfo()['minrelaytxfee'], Decimal('0.00001000'))
+        assert_equal(node.getmempoolinfo()['mempoolminfee'], Decimal('0.00001000'))
 
-        txids = []
-        utxos = create_confirmed_utxos(relayfee, self.nodes[0], 91)
+        tx_ids=[]
+        miniwallet.generate(1 + (3*30) + 1)
+        # We generate 92 Coinbase UTXOs here because we need
+        # 1 to create a tx that will be evicted from the mempool later
+        # 30 with a fee higher than the previous UTXO
+        # Another 30 with a fee higher than the previous set of UTXOs
+        # Another 30 with a fee even higher than  the previous set of UTXOs
+        # And 1 more to verify that this tx does not get added to the mempool with a relayfee (fee less than the mempoolminfee)
+
+        node.generate(COINBASE_MATURITY - 1)
+        # We mine 99 blocks so that we are allowed to spend our UTXOs
+        
 
         self.log.info('Create a mempool tx that will be evicted')
-        us0 = utxos.pop()
-        inputs = [{ "txid" : us0["txid"], "vout" : us0["vout"]}]
-        outputs = {self.nodes[0].getnewaddress() : 0.0001}
-        tx = self.nodes[0].createrawtransaction(inputs, outputs)
-        self.nodes[0].settxfee(relayfee) # specifically fund this tx with low fee
-        txF = self.nodes[0].fundrawtransaction(tx)
-        self.nodes[0].settxfee(0) # return to automatic fee selection
-        txFS = self.nodes[0].signrawtransactionwithwallet(txF['hex'])
-        txid = self.nodes[0].sendrawtransaction(txFS['hex'])
-
-        relayfee = self.nodes[0].getnetworkinfo()['relayfee']
-        base_fee = relayfee*100
-        for i in range (3):
-            txids.append([])
-            txids[i] = create_lots_of_big_transactions(self.nodes[0], txouts, utxos[30*i:30*i+30], 30, (i+1)*base_fee)
+        tx_to_be_evicted_id = miniwallet.send_self_transfer(from_node = node, fee_rate = relayfee)['txid']
+
+        base_fee = relayfee*1000
+        # We increase the tx fee massively now to give these transactions a higher priority in the mempool
+
+        for batch_of_txid in range (3):
+            tx_ids.append([])
+            tx_ids[batch_of_txid] = self.create_large_transactions(node, txouts, miniwallet, 30, (batch_of_txid+1)*base_fee)
```

Comment:
I know this was a relic of the existing test, but this data structure doesn't need to be so complex- looks like its only used for counting the total `tx_ids`, so you can simplify this by simply maintaining a count. 

that will also simplify line 63